src/Pagination.js

```diff
@@ -6,23 +6,30 @@ import PaginationButton from './PaginationButton';
 class Pagination extends Component {
   constructor (props) {
     super(props);
+    const { activePage, items } = props;
     this.state = {
-      activePage: this.props.activePage
+      activePage: activePage > 0 && activePage <= items
+        ? activePage : 1
     };
 
     this.renderButtons = this.renderButtons.bind(this);
     this._onClick = this._onClick.bind(this);
   }
 
+  componentWillReceiveProps ({ activePage }) {
+    const { items } = this.props;
```

Comment:
For now, I think we can merge this as is. I can't think of a scenario where the items would update externally.